profitbricks/src/test/java/org/jclouds/profitbricks/features/DataCenterApiMockTest.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.features;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+import java.util.List;
+
+import org.jclouds.profitbricks.ProfitBricksApi;
+import org.jclouds.profitbricks.domain.DataCenter;
+import org.jclouds.profitbricks.domain.Location;
+import org.jclouds.profitbricks.domain.ProvisioningState;
+import org.jclouds.profitbricks.internal.BaseProfitBricksMockTest;
+import org.jclouds.rest.ResourceNotFoundException;
+import org.testng.annotations.Test;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+/**
+ * Mock tests for the {@link org.jclouds.profitbricks.features.DataCenterApi} class
+ */
+@Test(groups = "unit", testName = "DataCenterApiMockTest")
+public class DataCenterApiMockTest extends BaseProfitBricksMockTest {
+
+   @Test
+   public void testGetAllDataCenters() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/datacenter/datacenters.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      DataCenterApi api = pbApi.dataCenterApi();
+
+      try {
+         List<DataCenter> dataCenters = api.getAllDataCenters();
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(dataCenters);
+         assertEquals(dataCenters.size(), 2);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
```

Comment:
Add also a test that verifies what happens when the server returns a 404. There is an explicit fallback in the method definition, so we should test it is taking place when needed.